### PR TITLE
added DELETE endpoint for LTGs

### PR DESCRIPTION
**Pull request by pmirski**
Currently requires input as query parameter "id"
e.g. POST http://localhost:3000/goals/deleteltg?id=5fa83412bc8b815b7885cbbf

### CODE REVIEW

**Comment by pmirski** [comment]
Made changes such that, now, 
1. method in back end code is called deleteLTG
2. example API call:
DELETE http://localhost:3000/goals/5fa83412bc8b815b7885cbbf

File: backend/src/goals/goals.js [hunk 1]
```diff
@@ -266,11 +266,40 @@ const completeShortTermGoal = async (req, res) => {
   }
 };
 
+const deleteLTG = async (req, res) => {
+  const ltgId = req.params;
+
+  if (ltgId == null) {
+    logger.info("Missing parameters in ${JSON.stringify(req.query)}.");
```
Comment (yaoharry) [inline backend/src/goals/goals.js:273]: quick thing, just change req.query to req.params here.

File: backend/src/goals/goals.js [hunk 2]
```diff
@@ -266,11 +266,40 @@ const completeShortTermGoal = async (req, res) => {
...
+  }
+};
+
 router.get("/", auth.checkIfAuthenticated, getGoals);
 router.post("/", auth.checkIfAuthenticated, postGoal);
 router.put("/:ltgId", auth.checkIfAuthenticated, updateGoal);
 router.get("/shortterm", auth.checkIfAuthenticated, getShortTermGoals);
 router.get("/suggestedstg", auth.checkIfAuthenticated, getSuggestedShortTermGoal);
 router.put("/shortterm/counter", auth.checkIfAuthenticated, completeShortTermGoal);
+router.delete("/:_id", auth.checkIfAuthenticated, deleteLTG);
```
Comment (yaoharry) [inline backend/src/goals/goals.js:303]: shouldn't this be /:ltgId ?

File: backend/src/goals/goals.js [hunk 3]
```diff
@@ -266,11 +266,40 @@ const completeShortTermGoal = async (req, res) => {
...
+
+  if (ltgId == null) {
+    logger.info("Missing parameters in ${JSON.stringify(req.query)}.");
+    res.status(400);
+    res.end();
+    return;
+  }
+
+  try {
+    GoalModel.findOneAndDelete(ltgId, function (err, res) {
```
Comment (yaoharry) [inline backend/src/goals/goals.js:280]: Is ltgId already a JSON object?

**Comment by yaoharry** [comment]
I'm not sure if we discussed if we wanted to outright delete the object from the database or simply set a flag that it has been deleted? This is mostly a design choice, as we wanted to have the ability to view old long term goals in the future? @jeremywiens @n-lina any thoughts?

File: backend/src/goals/goals.js (hunk 3 above)
Comment (pmirski) [reply backend/src/goals/goals.js:280]: With most recent commit, it is.

File: backend/src/goals/goals.js (hunk 2 above)
Comment (pmirski) [reply backend/src/goals/goals.js:303]: I don't think so. Doing it this way ensures that 
ltgId = req.params;
means that ltgId is an object of the format
{"_id":"ewkjdabflsakjdfb"}
and so the search is by property "_id"

**Pull request merged**